soh/soh/OTRGlobals.cpp

```diff
@@ -1766,10 +1771,14 @@ extern "C" int CustomMessage_RetrieveIfExists(GlobalContext* globalCtx) {
     char* buffer = font->msgBuf;
     const int maxBufferSize = sizeof(font->msgBuf);
     CustomMessageEntry messageEntry;
+    Player* player = GET_PLAYER(globalCtx);
     if (gSaveContext.n64ddFlag) {
         if (textId == TEXT_RANDOMIZER_CUSTOM_ITEM) {
-            messageEntry =
-                Randomizer_GetCustomGetItemMessage(GET_PLAYER(globalCtx));
+            if (player->getItemEntry.getItemId == RG_ICE_TRAP) {
+                messageEntry = Randomizer_GetIceTrapMessage();
+            } else {
+                messageEntry = Randomizer_GetCustomGetItemMessage(player);
+            }
```

Comment:
since we're not using `player` unless we get to the `textId == TEXT_RANDOMIZER_CUSTOM_ITEM` block, it probably makes sense to move the `Player* player = GET_PLAYER(globalCtx);` down into there

Reply:
Good catch, I'll change this.